src/Microsoft.AspNet.Server.Kestrel/Http/SocketOutput.cs

```diff
@@ -440,7 +481,11 @@ public void DoWriteIfNeeded()
                 }, this);
 
                 Self._head = _lockedEnd.Block;
-                Self._head.Start = _lockedEnd.Index;
```

Comment:
**Wait a minute!** I think this race must have been caused by a change in this PR

I was wondering how I didn't see this was a race when I originally wrote this code, and now I'm pretty sure it's because `_head` is only ever supposed to modified in the SocketOutput ctor and on the libuv thread. If that's the case, this race should be impossible.

@benaadams Do you have any idea how `_head` would be modified off of the libuv thread with this change? I'm not seeing it yet.